domain/secret/state/state.go

```diff
@@ -82,16 +82,17 @@ func (st State) getModelUUID(ctx context.Context, tx *sqlair.TX) (string, error)
 // [secreterrors.SecretAlreadyExists]if a user secret with the same
 // label already exists.
 func (st State) CreateUserSecret(
-	ctx context.Context, version int, uri *coresecrets.URI, secret domainsecret.UpsertSecretParams,
-) error {
+	ctx context.Context, version int, uri *coresecrets.URI,
+	secret domainsecret.UpsertSecretParams,
+) (string, error) {
 	db, err := st.DB()
 	if err != nil {
-		return errors.Trace(err)
+		return "", errors.Trace(err)
 	}
 
 	revisionUUID, err := uuid.NewUUID()
```

Comment:
Should this not be a string, but instead a typed secret revision? That way we can ensure the type, and with the bonus we can validate the type at any layer.

Reply:
After discussed with Ian, we will probably do this in future PR once we agreed to change all the UUID PKs to typed IDs.